packages/flutter/lib/src/widgets/two_dimensional_viewport.dart

```diff
@@ -848,11 +848,7 @@ abstract class RenderTwoDimensionalViewport extends RenderBox implements RenderA
     RenderBox? child = _firstChild;
     while (child != null) {
       final TwoDimensionalViewportParentData childParentData = parentDataOf(child);
-      // TODO(Piinks): When ensure visible is supported, remove this isVisible
-      //  condition.
-      if (childParentData.isVisible) {
-        visitor(child);
-      }
+      visitor(child);
```

Comment:
Do you know what semantics order the children in cache extent will be?

The semantics tree will use the screen order to determine the child order, but I remember the cached children will not have rects. I guess it is probably using the rendering child order for 1D case, but I am not sure.

Reply:
They will be in the same traversal order as the children not in the cache extent. In 2D, after layout, all the children are reified so that regardless of what order the subclass did layout in, they are in the correct order for paint and semantics. This is dictated by the mainAxis, which by default (Axis.vertical) reads in row-major order.